### ISSUE TITLE
Nix depends on openssl.bin

### ISSUE THREAD

**Issue by catern** [NixOS/nixpkgs#120948]
**Describe the bug**
openssl.dev depends on openssl.bin which bloats the closure size of Nix among other things.

**To Reproduce**
```
~/.local/src/nixpkgs $ nix-store --query --requisites $(nix-build -A openssl.dev) | grep openssl
/nix/store/5bg9dy337rp8r1mcc4cdrjwhw1kwmppj-openssl-1.1.1k
/nix/store/w99zrfcvxvh0y46sx8p6lbdjla4wdxnj-openssl-1.1.1k-bin
/nix/store/8iyzxj2ysxnknxc7jrldcsk2zzhv7ff1-openssl-1.1.1k-dev
```

**Expected behavior**
Not see -bin in the dependencies of openssl.

**Notify maintainers**
@peti 
@thefloweringash 
I believe this was introduce in https://github.com/NixOS/nixpkgs/commit/ef24a2815

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: openssl
# a list of nixos modules affected by the problem
module:
```

**Comment by hmenke** [comment]
That's by design.

https://github.com/NixOS/nixpkgs/blob/3bed46e4b91ec1c55bd614ca818dac0bdb0dda01/pkgs/build-support/setup-hooks/multiple-outputs.sh#L167-L199

**Comment by hmenke** [comment]
This should fix it (untested)
```patch
diff --git a/pkgs/development/libraries/openssl/default.nix b/pkgs/development/libraries/openssl/default.nix
index 13cac47793e..30e78d712e2 100644
--- a/pkgs/development/libraries/openssl/default.nix
+++ b/pkgs/development/libraries/openssl/default.nix
@@ -153,6 +153,8 @@ let
       rmdir $out/etc/ssl/{certs,private}
     '';
 
+    propagatedBuildOutputs = [ "out" "lib" ];
+
     postFixup = lib.optionalString (!stdenv.hostPlatform.isWindows) ''
       # Check to make sure the main output doesn't depend on perl
       if grep -r '${buildPackages.perl}' $out; then
```

**Comment by catern** [comment]
Hm, yeah, maybe dev depending on bin is intended, but somewhere along the line something must be wrong - I don't think it's intended for Nix to depend on openssl.bin:
```
~/.local/src/nixpkgs $ nix why-depends -f . nix openssl.bin
/nix/store/pva2sb1i6szpc4sc9a5pbc09ag6kflk4-nix-2.3.10
╚═══lib/libnixstore.so: …eccomp-2.5.1-lib/lib:/nix/store/qfmrq5nax53dhv6kvnhx6fsqlbjq22az-aws-sdk-cpp-1.8.121/lib:/nix/st…
    => /nix/store/qfmrq5nax53dhv6kvnhx6fsqlbjq22az-aws-sdk-cpp-1.8.121
    ╚═══lib/libaws-cpp-sdk-core.so: …yjjh-curl-7.74.0/lib:/nix/store/dr03rcg6d37f0x0y088rhkh6wl7ascdy-s2n-tls-1.0.0/lib:/nix/store/p5…
        => /nix/store/dr03rcg6d37f0x0y088rhkh6wl7ascdy-s2n-tls-1.0.0
        ╚═══nix-support/propagated-build-inputs: …/nix/store/8iyzxj2ysxnknxc7jrldcsk2zzhv7ff1-openssl-1.1.1k-dev …
            => /nix/store/8iyzxj2ysxnknxc7jrldcsk2zzhv7ff1-openssl-1.1.1k-dev
            ╚═══nix-support/propagated-build-inputs: … /nix/store/w99zrfcvxvh0y46sx8p6lbdjla4wdxnj-openssl-1.1.1k-bin /nix/store/5…
                => /nix/store/w99zrfcvxvh0y46sx8p6lbdjla4wdxnj-openssl-1.1.1k-bin
```

**Comment by hmenke** [comment]
> maybe dev depending on bin is intended

I don't think it is intended. It's just the default which was not changed here. In fact, when I inspected `dev` I did not find any other reference to `bin` than in `nix-support/propagated-build-inputs`. So I think it should be safe to remove `bin` from `propagatedBuildOutputs` like in the above patch. Unfortunately, properly testing this would trigger a rebuild of pretty much *all* of nixpkgs.
